walkthroughs/week-5-tdd/project/src/main/java/com/google/sps/FindMeetingQuery.java

```diff
@@ -15,9 +15,117 @@
 package com.google.sps;
 
 import java.util.Collection;
+import java.util.Arrays;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.stream.Collectors;
 
 public final class FindMeetingQuery {
   public Collection<TimeRange> query(Collection<Event> events, MeetingRequest request) {
-    throw new UnsupportedOperationException("TODO: Implement this method.");
+    if(events.isEmpty() && request.getDuration() <= TimeRange.WHOLE_DAY.duration()) {
+      return Arrays.asList(TimeRange.WHOLE_DAY);
+    }
+
+    if(request.getDuration() > TimeRange.WHOLE_DAY.duration()) {
+      return Arrays.asList();
+    }
+    
+    List<TimeRange> availableTimes = new ArrayList<TimeRange>();
+    List<Event> eventList = new ArrayList<>(events);
+    Event firstEvent = eventList.get(0);
+    Event lastEvent = eventList.get(eventList.size() - 1);
+
+    int eventsSkipped = 0;
+
+    if(eventList.size() == 1) {
+      if(!requestHasAttendees(request, firstEvent)) {
+        eventsSkipped++;
+      }
+    }
+
+    // Add the first time slot, from the start of the day to the start of the first event,
+    // only if there's no event that starts the day.
+    if(dayStartsWithEvent(eventList) && requestHasAttendees(request, firstEvent)) {
+      availableTimes.add(TimeRange.fromStartEnd(TimeRange.START_OF_DAY,
+        eventList.get(0).getWhen().start(), false));
+    }
+
+    for (int i = 0; i < eventList.size() - 1; i++) {
+      TimeRange currEventTimeRange = eventList.get(i).getWhen();
+      TimeRange nextEventTimeRange = eventList.get(i + 1).getWhen();
+
+      if(!requestHasAttendees(request, currEvent)) {
+        eventsSkipped ++;
+        continue;
+      }
+
+      if(currEventTimeRange.contains(nextEventTimeRange)) {
+        // If there is a 3rd event, connect events 1 and 3. Otherwise, connect with the end of day.
+        if(i + 2 < eventList.size() - 1) {
+          TimeRange nextnextEventTimeRange = eventList.get(i + 2).getWhen();
+          availableTimes.add(TimeRange.fromStartEnd(currEventTimeRange.end(),
+            nextnextEventTimeRange.start(), true));
+          i++;
+        } else {
+          availableTimes.add(TimeRange.fromStartEnd(currEventTimeRange.end(),
+            TimeRange.END_OF_DAY, true));
+        }
+      } else {
+          if(currEventTimeRange.overlaps(nextEventTimeRange)) {
+            continue;
```

Comment:
Add an inline comment explaining why we break the flow here

Reply:
I don't understand this case here. I think that what you need to do depends on whether the nextEvent has any of the requested attendees